src/main/java/duke/commands/Command.java

```diff
@@ -0,0 +1,65 @@
+package duke.commands;
+
+import java.time.LocalDate;
+
+/**
+ * Represents a command to be executed.
+ */
+public class Command {
+    private final String commandType;
+    private String description;
+    private int taskIndex;
+    private LocalDate deadlineDate;
+    private LocalDate eventFromDate;
+    private LocalDate eventToDate;
+
+    public Command(String commandType) {
+        this.commandType = commandType;
+    }
+
+    public Command(String commandType, String description) {
```

Comment:
Rather than creating all these different public constructors, would it be easier to write an Of factory method and change these constructors to private?